lib/ConstExtract/ConstExtract.cpp

```diff
@@ -128,69 +128,116 @@ static std::string extractLiteralOutput(Expr *expr) {
   return LiteralOutput;
 }
 
-static std::shared_ptr<CompileTimeValue>
-extractPropertyInitializationValue(VarDecl *propertyDecl) {
-  auto binding = propertyDecl->getParentPatternBinding();
-  if (binding) {
-    auto originalInit = binding->getOriginalInit(0);
-    if (originalInit) {
-      auto literalOutput = extractLiteralOutput(originalInit);
-      if (!literalOutput.empty()) {
-        return std::make_shared<RawLiteralValue>(literalOutput);
-      }
+static std::shared_ptr<CompileTimeValue> extractCompileTimeValue(Expr *expr) {
+  switch (expr->getKind()) {
+  case ExprKind::Array:
+  case ExprKind::Dictionary:
+  case ExprKind::BooleanLiteral:
+  case ExprKind::IntegerLiteral:
+  case ExprKind::FloatLiteral:
+  case ExprKind::NilLiteral:
+  case ExprKind::StringLiteral: {
+    std::string literalOutput;
+    llvm::raw_string_ostream OutputStream(literalOutput);
+    expr->printConstExprValue(&OutputStream, nullptr);
+
+    if (literalOutput.empty()) {
+      return std::make_shared<RuntimeValue>();
+    }
 
-      if (auto callExpr = dyn_cast<CallExpr>(originalInit)) {
-        if (callExpr->getFn()->getKind() != ExprKind::ConstructorRefCall) {
-          return std::make_shared<RuntimeValue>();
-        }
+    return std::make_shared<RawLiteralValue>(literalOutput);
+  }
 
-        std::vector<FunctionParameter> parameters;
-        const auto args = callExpr->getArgs();
-        for (auto arg : *args) {
-          auto label = arg.getLabel().str().str();
-          auto expr = arg.getExpr();
+  case ExprKind::Tuple: {
+    auto tupleExpr = cast<TupleExpr>(expr);
 
-          switch (expr->getKind()) {
-          case ExprKind::DefaultArgument: {
-            auto defaultArgument = cast<DefaultArgumentExpr>(expr);
-            auto *decl = defaultArgument->getParamDecl();
+    std::vector<TupleElement> elements;
+    if (tupleExpr->hasElementNames()) {
+      for (auto pair :
+           llvm::zip(tupleExpr->getElements(), tupleExpr->getElementNames())) {
+        auto elementExpr = std::get<0>(pair);
+        auto elementName = std::get<1>(pair);
 
-            if (decl->hasDefaultExpr()) {
-              literalOutput =
-                  extractLiteralOutput(decl->getTypeCheckedDefaultExpr());
-            }
+        Optional<std::string> label =
+            elementName.empty()
+                ? Optional<std::string>()
+                : Optional<std::string>(elementName.str().str());
 
-            break;
-          }
-          default:
-            literalOutput = extractLiteralOutput(expr);
-            break;
-          }
+        elements.push_back({label, elementExpr->getType(),
+                            extractCompileTimeValue(elementExpr)});
+      }
+    } else {
+      for (auto elementExpr : tupleExpr->getElements()) {
+        elements.push_back({Optional<std::string>(), elementExpr->getType(),
+                            extractCompileTimeValue(elementExpr)});
+      }
```

Comment:
@artemcm - I wasn't able to exercise this codepath. Do you happen to know under what circumstances `tupleExpr->hasElementNames()` would be false?

I had thought `let t1: (String, Bar) = ("foo", Bar())` would cause that, but instead the tuple expression's identifier is `.empty()`.

Reply:
Looks like the parser sets things up with empty identifiers rather than no identifiers. 
One example I see where `hasElementNames()` would return `false` is an empty tuple. And it's possible there is other places where the compiler itself synthesizes a tuple where labels are omitted. So it should be fine to keep this code-path as-is for safety.